src/analysis/normed_space/weak_dual_of_normed_space.lean

```diff
@@ -0,0 +1,163 @@
+/-
+Copyright (c) 2021 Kalle Kytölä. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Kalle Kytölä and Heather Macbeth
+-/
+import tactic
+import topology.algebra.weak_dual_topology
+import analysis.normed_space.dual
+import analysis.normed_space.operator_norm
+
+noncomputable theory
+open filter
+open_locale topological_space
+
+section weak_star_topology_for_duals_of_normed_spaces
+
+/-!
+### Weak star topology on duals of normed spaces
+In this section, we prove properties about the weak-* topology on duals of normed spaces.
+We prove in particular that the canonical mapping `dual 𝕜 E → weak_dual 𝕜 E` is continuous,
+i.e., that the weak-* topology is coarser (not necessarily strictly) than the topology given
+by the dual-norm (i.e. the operator-norm).
+-/
+
+open normed_space
+
+variables {𝕜 : Type*} [nondiscrete_normed_field 𝕜]
+variables {E : Type*} [normed_group E] [normed_space 𝕜 E]
+
+/-- For normed spaces `E`, there is a canonical map `dual 𝕜 E → weak_dual 𝕜 E`. -/
+def normed_space.dual.to_weak_dual : dual 𝕜 E → weak_dual 𝕜 E := id
```

Comment:
You can streamline the next 30 lines or so by using a more souped-up `id` to start with:
```suggestion
/-- For normed spaces `E`, there is a canonical map `dual 𝕜 E ≃ₗ[𝕜] weak_dual 𝕜 E`. -/
def normed_space.dual.to_weak_dual : dual 𝕜 E ≃ₗ[𝕜] weak_dual 𝕜 E :=
linear_equiv.refl 𝕜 (E →L[𝕜] 𝕜)
```

Reply:
By "streamline" you mean delete altogether (with perhaps 2 exceptions), right? ;)

Done. Much better (by approx -30 lines) now.